File: frontend/src/types/attackLayer.ts
export interface LayerEntry {
  technique_id: string
  name: string
  color: string
  comment: string
  score: number
  enabled: boolean
}

export interface AttackLayer {
  id: number
  name: string
  description: string
  domain: string
  version: string
  layer_data: string
  created_at: string
  updated_at: string
}

export interface AttackLayerListResponse {
  items: AttackLayer[]
  total: number
}

export interface AttackLayerCreatePayload {
  name: string
  description?: string
  domain?: string
  version?: string
}

export interface NavigatorExport {
  name: string
  description: string
  domain: string
  versions: Record<string, string>
  techniques: {
    techniqueID: string
    tactic?: string
    color: string
    comment: string
    score: number
    enabled: boolean
  }[]
}

File: backend/internal/models/attack_layer.go
package models

import "time"

type AttackLayer struct {
	ID          uint      `json:"id"          gorm:"primaryKey;autoIncrement"`
	Name        string    `json:"name"        gorm:"type:text;not null"`
	Description string    `json:"description" gorm:"type:text"`
	Domain      string    `json:"domain"      gorm:"type:text;not null;default:'enterprise-attack'"`
	Version     string    `json:"version"     gorm:"type:text;not null;default:'4.5'"`
	LayerData   string    `json:"layer_data"  gorm:"type:text"` // JSON array of LayerEntry
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}
